py/picca/pk1d/postproc_pk1d.py

```diff
@@ -79,66 +61,74 @@ def read_pk1d(filename, kbin_edges, snrcut=None, zbins_snrcut=None):
     """
     p1d_table = []
     z_array = []
-    with fitsio.FITS(filename, 'r') as hdus:
+    with fitsio.FITS(filename, "r") as hdus:
         for i, hdu in enumerate(hdus[1:]):
             data = hdu.read()
             chunk_header = hdu.read_header()
             chunk_table = Table(data)
             for colname in [
-                    'k', 'Pk', 'Pk_raw', 'Pk_noise', 'Pk_diff', 'cor_reso',
-                    'Pk_noise_miss'
+                "k",
+                "Pk",
+                "Pk_raw",
+                "Pk_noise",
+                "Pk_diff",
+                "cor_reso",
+                "Pk_noise_miss",
             ]:
                 try:
                     chunk_table.rename_column(colname.upper(), colname)
                 except KeyError:
                     pass
 
-            if np.nansum(chunk_table['Pk']) == 0:
-                chunk_table['Pk'] = (
-                    chunk_table['Pk_raw'] -
-                    chunk_table['Pk_noise']) / chunk_table['cor_reso']
+            if np.nansum(chunk_table["Pk"]) == 0:
+                chunk_table["Pk"] = (
+                    chunk_table["Pk_raw"] - chunk_table["Pk_noise"]
+                ) / chunk_table["cor_reso"]
 
-            chunk_table['forest_z'] = float(chunk_header['MEANZ'])
-            chunk_table['forest_snr'] = float(chunk_header['MEANSNR'])
+            chunk_table["forest_z"] = float(chunk_header["MEANZ"])
+            chunk_table["forest_snr"] = float(chunk_header["MEANSNR"])
+            chunk_table["forest_id"] = int(chunk_header["LOS_ID"])
+            if "CHUNK_ID" in chunk_header:
+                chunk_table[
+                    "chunk_id"
+                ] = f"{chunk_header['LOS_ID']}_{chunk_header['CHUNK_ID']}"
```

Comment:
You mean los_id has not been part of the files or was just ignored, probably the latter. I don't really have a strong opinion regarding using chunk_id to simplify matching, just wanted to note we would have enough info even without.
I guess my intuition that chunks are roughly the same size as zbins failed me here, this is only actually true for z>~2.8, so it should be fine to have this often at lower redshifts.
In principle, you wouldn't need to know anything about the LOS_ID or CHUNK_ID, just if the spectrum is usable and if it falls in the redshift bin of interest as in both, you're averaging over all combinations of chunks anyway later, could be useful for approximations though

Reply:
I do not agree with your last statement. If we use all chunk_id in one redshift bin, it would supress the signal as we would compute the cross-correlation between different forest (which might be far also) and we want to compute the covariance for P1D not P3D. 
los_id was previously ignored

Reply:
But the cross-correlation between far away forests is something we intrinsically have as part of the P1d measurements as P1d is something different than P3d (kperp~0).

In my view it should be roughly like the following:
- for the bootstrap: you calculate the C_ij = <P_full_set(ki,zi) P_full_set(kj,zj)>_{bootstrap realizations} where indices i,j go over all combinations of k,z that you want to consider. The avg here is over all different bootstrap samples and the P_full_set are computed as the <P> over all chunks contributing to that bin (potentially with some weighting), having chunk_id and los_id simplifies this
- for the sub-sample covariance (with sample size==1): you take all the forests providing information to those bin combinations and compute C_ij = <P_single(ki,zi) P_single(kj,zj)>/(Nforests-1). Also here it would be enough to go through all spectra, test if they contribute to ki, zi, kj, zj and put them into that average. I guess it can be useful to have an index, but it's not required. (Actually I'm not sure anymore how one would do the multi-z correlations this way as there are currently no forests contributing to two different redshift bins, for this purpose one might want to consider either subsamples of size>1 or the different chunks of same spectra falling into different z-bins [which then requires the indexing])
- to speed things up I think one can do the k_binning of the individual forests beforehand which then requires you to store the number of modes going into that allowing to weight each mode the same way